core/src/main/java/com/sequenceiq/distrox/v1/distrox/controller/DistroXUpgradeV1Controller.java

```diff
@@ -76,8 +76,7 @@ public DistroXRdsUpgradeV1Response upgradeRdsByName(@ResourceName String name, @
 
     @Override
     @CheckPermissionByResourceCrn(action = AuthorizationResourceAction.UPGRADE_DATAHUB)
-    public DistroXRdsUpgradeV1Response upgradeRdsByCrn(@ValidCrn(resource = CrnResourceDescriptor.DATAHUB)
```

Comment:
validation for datahub should be kept here, since controller and api indicates that this is only for datahub, if an API needed for DL and DH too, different API should be implemented for that purpose

Reply:
the current setup looks good:
datalake API should be used for DL (SdxUpgradeController)
distrox API should be used for DH (this class)
for internal call, StackV4Controller#upgradeRdsByClusterNameInternal can be used for both

@gergopapi2 what do you think?

Reply:
agree, this validation should be kept in place

Reply:
Thats what is happening. I only removed @ValidCrn(resource = CrnResourceDescriptor.DATAHUB) from this class since it's already present in corresponding declaration of this method in DistroXUpgradeV1Endpoint. Due to this, It was being validated twice causing incorrect error message so I fixed the same by keeping the annotation in interface only.

Reply:
oh, i see, then this should be fine